pallets/parachain-staking/src/lib.rs

```diff
@@ -2252,39 +2449,40 @@ pub mod pallet {
 			candidate: T::AccountId,
 			delegator: T::AccountId,
 		) {
-			if let Ok(amount_transferred) =
-				T::Currency::deposit_into_existing(&delegator, amt.clone())
-			{
-				Self::deposit_event(Event::Rewarded {
-					account: delegator.clone(),
-					rewards: amount_transferred.peek(),
-				});
+			// Mint rewards to the delegator
+			if frame_system::Pallet::<T>::account_exists(&delegator) {
```

Comment:
I think this check is not needed.

Reply:
It does not check internally.

Reply:
Yes, but if we think about it, the check is not necessary, the mint_into already mutates the account even if the account does not exist.

Reply:
Which at this point must already exist.

Reply:
Fair